popsycle/synthetic.py

```diff
@@ -3512,3 +3527,402 @@ def generate_slurm_scripts(slurm_config_filename, popsycle_config_filename,
+    l_min = np.min(long_bin).values
+    l_max = np.max(long_bin).values
+    b_min = np.min(lat_bin).values
+    b_max = np.max(lat_bin).values
+
+    dists = angdist(l, b, c.l.deg, c.b.deg)
+
+    #Masking the full PBH data for our field of view
+    mask = (c.distance.kpc <= 2*r_max) & (dists < field_of_view_diameter)
+    data_in_field = c[mask]
+
+    #Obtain the radius values for all of the PBHs in our field of view.
+    r_in_field = data_in_field.distance.kpc
+    l_in_field = data_in_field.l.deg
+    b_in_field = data_in_field.b.deg
+
+    c_pbh = coord.Galactic(l=l_in_field * units.deg, b=b_in_field * units.deg, distance=r_in_field * units.kpc)
+    c_pbh = c_pbh.transform_to(coord.Galactocentric(representation_type='spherical'))
+    cart_pbh = astropy.coordinates.cartesian_to_spherical(c_pbh.x, c_pbh.y, c_pbh.z)
+    pbh_r_galacto = cart[0]
+
+    if inner_slope == 1:
+        vel_data = pd.read_csv('data/inner_slope_one.csv')
+    elif inner_slope == .25:
+        vel_data = pd.read_csv('data/inner_slope_quarter.csv')
+    else:
+        vel_data = pd.read_csv('data/inner_slope_half.csv')
+
+    pbh_vrms = np.interp(pbh_r_galacto, vel_data['x'], vel_data['y'])
+    v_vals = np.arange(0, v_esc) #Goes from v to v_esc
+    a = (1/2)*pbh_vrms*((np.pi/2)**(1/2))
+
+    rand_cdf = []
+
+    for a_val in a:
+        pdf = ((2/np.pi)**(1/2))*((v_vals**2*np.exp(-v_vals**2/(2*a_val**2)))/a_val**3)
+        cdf = scipy.special.erf(v_vals/(a_val*2**(1/2)))-(((2/np.pi)**(1/2))*((v_vals*np.exp(-v_vals**2/2*a_val**2))/a_val))
+        rand_cdf.append(np.random.uniform(0, np.amax(cdf), 1))
+    interpreted_rms_velocities = np.interp(rand_cdf, cdf, v_vals)
+
+    #Sampling random latitude and longitude values for velocity to complete the spherical velocities
+    sin_lat_vel = np.random.uniform(-1, 1, len(data_in_field))
+    lat_vel = np.arcsin(sin_lat_vel)
+    long_vel = np.random.uniform(0, 2*np.pi, len(data_in_field))
+
+    #Transforming velocities to cartesian to get vx, vy, vz values
+    cart_vel = astropy.coordinates.spherical_to_cartesian(interpreted_rms_velocities, lat_vel, long_vel)
+    vx= cart_vel[0]
+    vy= cart_vel[1]
+    vz= cart_vel[2]
+
+    #Getting rest of data for PBHs for the combined .h5 file
+    mass = np.full(len(data_in_field), pbh_mass)
+    zams_mass = mass
+    age = np.full(len(data_in_field), 14*(10**9))
+    pop_id = np.full(len(data_in_field), 10)
+    rem_id = np.full(len(data_in_field), 104)
+
+    b_rad = np.radians(b_in_field)
+    l_rad = np.radians(l_in_field)
+
+    cart_helio = astropy.coordinates.spherical_to_cartesian(r_in_field, b_rad, l_rad)
+
+    px = cart_helio[0]
+    py = cart_helio[1]
+    pz = cart_helio[2]
+
+    vr, mu_b, mu_lcosb = calc_sph_motion(vx, vy, vz, r_in_field, b_in_field, l_in_field)
+
+    exbv = np.full(len(data_in_field), 0)
+    ubv_k = np.full(len(data_in_field), np.nan)
+    ubv_j = np.full(len(data_in_field), np.nan)
+    obj_id = np.arange(1, (len(data_in_field)+1))
```

Comment:
Will this object id collide with others in the dataset? I think that it will. Come up with a convention that makes sure that all object ids are unique.